DM-41650 Add step definitions to pipelines (#392)
lsst/pipe_base: doc/changes/DM-41650.feature.md

> Added a section to pipelines which allows the explicit declaration of which susbsets correspond to steps and what the dimensions that steps are expected to operate over.

Reviewer: ```suggestion
Added a section to pipelines which allows the explicit declaration of which susbsets correspond to steps and the dimensions the step's quanta can be sharded with.
```
